src/dgipy/vcf.py

```diff
@@ -0,0 +1,202 @@
+        # TODO: handle genes without names, 'novel transript'
+        try:
+            self.gene = data[0]["name"]
+        except:
+            self.gene = "None"
+
+        self.records = data
+        self.interactions = self.search_interactions(self.gene)
+
+        # TODO: handle app searches with blank lists [], currently FDA resource hangs for awhile?
+        if not list(self.interactions["drug"].values):
+            self.applications = "None"
+        else:
+            self.applications = self.search_applications(
+                list(self.interactions["drug"].values)
+            )
+
+        self.gene_info = self.grab_gene_info(self.gene)
+        self.categories = self.grab_categories(self.gene)
+
+    # TODO: These can probably just be simpilifed to direct assignment during __init__ but might be useful this way too
+    def search_interactions(self, gene: str) -> pd.DataFrame:
+        """Search drug-gene interactions for given gene in DGIdb
+
+        :param gene: the name of the gene
+        :return: Dataframe of drug-gene interactions
+        """
+        return dgipy.get_interactions(gene)
+
+    def search_applications(self, drugs: list) -> pd.DataFrame:
+        """Search for drug applications for interaction drugs in DGIdb
+
+        :param drugs: list of drugs from interaction results
+        :return: Dataframe of FDA applications
+        """
+        return dgipy.get_drug_applications(drugs)
+
+    def grab_gene_info(self, gene: str) -> pd.DataFrame:
+        """Grab gene information from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene information
+        """
+        return dgipy.get_gene(gene)
+
+    def grab_categories(self, gene: str) -> pd.DataFrame:
+        """Grab gene categories from DGIdb
+
+        :param gene: gene name
+        :return: Dataframe of gene categories
+        """
+        return dgipy.get_categories(gene)
+
+
+def annotate(filepath: str, contig: str) -> pd.DataFrame:
+    """Map chr,pos pairs from a VCF file to human genes and search DGIdb for drug-gene interactions
+
+    :param filepath: link to a valid VCF file
+    :param contig: specified chromosome (i.e. chr7)
+    :return: Dataframe of drug-gene interactions
+    """
+    # Open VCF file
+    records = _process_vcf(filepath, contig)
+    # Grab records & relevant info (params: chr7)
+    mapped = _ensembl_map(records)  # TODO: modularize mapping
+    # Group records with like-genes
+    grouped = _group_by_name(mapped)
+    # Instance each gene set as a class
+    vcf_results = []
+    for gene in grouped:
+        vcf_results.append(GeneResult(grouped[gene]))
+
+    return vcf_results
+
+
+def _process_vcf(filepath: str, contig: str) -> list:
+    """Grab relevant data for mapping and mutations from starting VCF
+
+    :param filepath: link to valid VCF file
+    :param contig: specified chromosome (i.e. chr7)
+    :return: List of record dicts
+
+    """
+    file = pysam.TabixFile(filepath)
```

Comment:
any reason why we aren't using `pysam.VariantFile`? https://pysam.readthedocs.io/en/latest/usage.html#working-with-vcf-bcf-formatted-files

Reply:
Biggest reason is just bc the first sample file I worked with was a vcf with an associated tabix file. Could easily modularize this in the future? Adding a TODO